File: leet-code/removeDuplicates.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function (nums) {
  let currIndex = 1
  let currVal = nums[0]
  for (let i = 1; i < nums.length; i++) {
    if (nums[i] !== currVal) {
      currVal = nums[i]
      nums[currIndex] = currVal
      currIndex++
    }
  }
  return currIndex
}

console.log(removeDuplicates([-1, 0, 0, 0, 0, 3, 3]))

File: leet-code/removeDuplicates.go
package main

func removeDuplicates(nums []int) int {
	if len(nums) == 0 {
		return 0
	}
	currIndex := 1
	currVal := nums[0]

	for i := 1; i < len(nums); i++ {
		if currVal != nums[i] {
			currVal = nums[i]
			nums[currIndex] = currVal
			currIndex++
		}
	}
	return currIndex
}
